# HTTP request to get LinkedIn access token
https://www.linkedin.com/oauth/v2/authorization?response_type=code&client_id=78eopeauzfd1my&redirect_uri=https://docs.microsoft.com/en-us/linkedin/shared/authentication/client-credentials-flow&scope=r_liteprofile&state=aRandomString
        
# Returned URL with code to get access token
https://docs.microsoft.com/en-us/linkedin/shared/authentication/client-credentials-flow?code=AQRc0pgDt8Af99wiMGmI-_SB4kvtc7sG2Q-HNu4vc2wQ9jBLxKFrNc18i361L5Wjsk4VDe6QSgVIJNKsbCeNIS9B8iTuSrJgtrxKEIu-deTyWOXVwKzzH2nr3v8Ql7HFrzJ6Iu7Tu0YeH0ADardJ--RQMxeMo03NdAUT_c-bbgauI3BHzl0a7GxC3_Upsw&state=aRandomString

In [ ]:
pip install python3-linkedin 

In [16]:
from linkedin import linkedin

# define application key and secret
APPLICATON_KEY    = '78eopeauzfd1my'
APPLICATON_SECRET = '8yZ6hunoNmJhCxmL'

# define return URL
RETURN_URL = 'https://docs.microsoft.com/en-us/linkedin/shared/authentication/client-credentials-flow'

# create authentication object
authentication = linkedin.LinkedInAuthentication(
                    APPLICATON_KEY,
                    APPLICATON_SECRET,
                    RETURN_URL,
                    linkedin.PERMISSIONS.enums.values()
                )

# using authorization code from HTTP request, get the access token (result)
authentication.authorization_code = 'AQRc0pgDt8Af99wiMGmI-_SB4kvtc7sG2Q-HNu4vc2wQ9jBLxKFrNc18i361L5Wjsk4VDe6QSgVIJNKsbCeNIS9B8iTuSrJgtrxKEIu-deTyWOXVwKzzH2nr3v8Ql7HFrzJ6Iu7Tu0YeH0ADardJ--RQMxeMo03NdAUT_c-bbgauI3BHzl0a7GxC3_Upsw'
result = authentication.get_access_token()

# print out the access token and its time of expiration -- one month from now
print ("Access Token:", result.access_token)
print ("Expires in (seconds):", result.expires_in)

Access Token: AQVj7APKloZ1BBPC0yZQ1Yei3Ow1i3v6pYSVfbk__QE6HKoSk1psnnm7wkuVi-39hRFBjlElSkNHUOWMZ8_RaNAmY_YTVQnfJiPWkqxR9FNAbVQoOyyRofjcv4l8ULhZ9_lMCmIM2c1rnxz5uLfHF5LmXgBT9_hH7WKA1M3qSChOOI4BAo7sMxNxIgutwKxLaMXlQef0dIDXbOBnC7xlv6J_X6PFtB9hqmdCNDkwrqYsSyX1WudmRYR7zwDkXW341aMBYuf141gQIa35ManPCt9mxZfD8EmssE6J3HMZj68uXh0LvqQazpOjlCweyMf-2Irikyrfu--7mpUc8rwkpmmXp-Kuyg
Expires in (seconds): 5183999


# Access Token

Access Token: AQVj7APKloZ1BBPC0yZQ1Yei3Ow1i3v6pYSVfbk__QE6HKoSk1psnnm7wkuVi-39hRFBjlElSkNHUOWMZ8_RaNAmY_YTVQnfJiPWkqxR9FNAbVQoOyyRofjcv4l8ULhZ9_lMCmIM2c1rnxz5uLfHF5LmXgBT9_hH7WKA1M3qSChOOI4BAo7sMxNxIgutwKxLaMXlQef0dIDXbOBnC7xlv6J_X6PFtB9hqmdCNDkwrqYsSyX1WudmRYR7zwDkXW341aMBYuf141gQIa35ManPCt9mxZfD8EmssE6J3HMZj68uXh0LvqQazpOjlCweyMf-2Irikyrfu--7mpUc8rwkpmmXp-Kuyg

Expires in (seconds): 5183999

In [10]:
from linkedin import linkedin

# function to share on LinkedIn
def post_To_LinkedIn():
    
    # define the application with the user access token
    application = linkedin.LinkedInApplication(token='AQVj7APKloZ1BBPC0yZQ1Yei3Ow1i3v6pYSVfbk__QE6HKoSk1psnnm7wkuVi-39hRFBjlElSkNHUOWMZ8_RaNAmY_YTVQnfJiPWkqxR9FNAbVQoOyyRofjcv4l8ULhZ9_lMCmIM2c1rnxz5uLfHF5LmXgBT9_hH7WKA1M3qSChOOI4BAo7sMxNxIgutwKxLaMXlQef0dIDXbOBnC7xlv6J_X6PFtB9hqmdCNDkwrqYsSyX1WudmRYR7zwDkXW341aMBYuf141gQIa35ManPCt9mxZfD8EmssE6J3HMZj68uXh0LvqQazpOjlCweyMf-2Irikyrfu--7mpUc8rwkpmmXp-Kuyg')
    
    application.submit_share('Posting from the API using JSON', 'A title for your share', None, 'http://www.linkedin.com', 'http://d.pr/3OWS')
    #application.submit_share(comment='Posting from Jupyter Notebooks', title='Test 1') 
    #application.submit_share("Test")
    #application.submit_share()

def submit_share(self, comment=None, title=None, description=None,
submitted_url=None, submitted_image_url=None,
visibility_code='anyone'):

In [11]:
post_To_LinkedIn()

AttributeError: 'HTTPError' object has no attribute 'message'